### DISCUSSION TITLE
Compilation error in HPC

### DISCUSSION THREAD

**Discussion by mehulbakhshi02** [su2code/SU2#2183] [General]
Hello, I am new to SU2 and trying to install it on cluster locally. I am facing issue in compilation and installation step.

I used following command to build:
`./meson.py -Dwith-mpi=enabled -Dcustom-mpi=true -Dextra-deps=mpich -Denable-cgns=false --prefix=$HOME/SU2/SU2-Install build`

After which I tried to compile and install:
`./ninja -C build install`

However I am getting an error:
`ninja: Entering directory build
[1/463] Compiling C object externals/metis/libmetis.a.p/GKlib_htable.c.o
FAILED: externals/metis/libmetis.a.p/GKlib_htable.c.o 
cc -Iexternals/metis/libmetis.a.p -Iexternals/metis -I../externals/metis -I../externals/metis/include -I../externals/metis/GKlib -I../externals/metis/libmetis -std=c99 -O3 -fPIC -D_FILE_OFFSET_BITS=64 -DNDEBUG -DNDEBUG2 -DHAVE_GETLINE -DUSE_GKREGEX -Wno-implicit-function-declaration -Wno-unused-result -Wno-unused-parameter -Wno-unused-variable -Wno-unused-but-set-variable -Wno-unknown-pragmas -Wno-sign-compare -Wno-clobbered -Wno-empty-body -Wno-unused-label -Wno-misleading-indentation -Wno-maybe-uninitialized -Wno-pedantic -Wno-error=absolute-value -MD -MQ externals/metis/libmetis.a.p/GKlib_htable.c.o -MF externals/metis/libmetis.a.p/GKlib_htable.c.o.d -o externals/metis/libmetis.a.p/GKlib_htable.c.o -c ../externals/metis/GKlib/htable.c
cc1: error: -Werror=absolute-value: no option -Wabsolute-value
cc1: warning: unrecognized command line option "-Wno-misleading-indentation" [enabled by default]`

I am unable to understand if its an issue with GCC compiler of the HPC cluster or something else. Would be a great help if someone can guide me on this. Thanks in advance!

**Cluster Information**
 - OS: [CentOS Linux 7]
 - C++ compiler and version: [g++ (GCC) 13.2.0] (Loaded as module)
 - MPI implementation and version: [mpich 4.1.2] (Loaded as module)
 - SU2 Version: [8.0.0 "Harrier"]

**Comment by jblueh** [comment]
It can happen that meson detects the system `gcc` instead of the one that is loaded as a module. In that case, setting the environment variables `CC` and `CXX` to point to the correct compiler executables does the trick. You can see the `gcc` that was detected in the meson output.

**Reply by mehulbakhshi02** [reply]
Thank you so much for the help!

SU2 was detecting c++ compiler but not c compiler. Giving appropriate CC and CXX paths did the work. Thank you very much.
